python/revng/api/_capi.py

```diff
@@ -0,0 +1,71 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+import traceback
+from pathlib import Path
+from typing import List
+
+from cffi import FFI
+from cffi.backend_ctypes import CTypesBackend
+
+from ..support import collect_libraries
+
+REVNG_ROOT_ENV = os.getenv("REVNG_ROOT")
+if REVNG_ROOT_ENV is not None:
+    ROOT = Path(REVNG_ROOT_ENV)
+else:
+    # We assume this file is in <root>/lib/python/revng/api
+    ROOT = (Path(__file__) / "../../../../..").resolve()
+HEADERS_DIR = ROOT / "include" / "revng" / "PipelineC"
+
+header_paths = [
+    HEADERS_DIR / "ForwardDeclarationsC.h",
+    HEADERS_DIR / "Prototypes.h",
+]
+
+ffi = FFI()
+for header_path in header_paths:
+    with open(header_path) as header_file:
+        lines = []
+        for line in header_file:
+            if not line.startswith("#"):
+                lines.append(line)
+        ffi.cdef("\n".join(lines))
+
+
+@ffi.callback("void ()")
+def pytraceback():
+    traceback.print_stack()
+
+
+LIB_DIR = ROOT / "lib"
+LIBRARY_PATH = LIB_DIR / "librevngPipelineC.so"
+
+assert LIBRARY_PATH.is_file()
+
+ARGV: List[str] = []
+
+REVNG_ANALYSIS_LIBRARIES = os.getenv("REVNG_ANALYSIS_LIBRARIES")
+if REVNG_ANALYSIS_LIBRARIES is not None:
+    libs = [Path(f) for f in REVNG_ANALYSIS_LIBRARIES.split(",")]
+    assert all(p.is_file() for p in libs), "Analysis libraries must all be present"
+    ANALYSIS_LIBRARIES = libs
+else:
+    collected_libs = collect_libraries([str(ROOT.resolve()), ])
+    ANALYSIS_LIBRARIES = [Path(p) for p in collected_libs[0]]
```

Comment:
Nice that we have a way to specify extra libs, but the defaults ones should be always loaded automatically, no matter what.